plugin/kubernetes/namespace.go

```diff
@@ -1,16 +1,19 @@
 package kubernetes
 
-// namespace checks if namespace n exists in this cluster. This returns true
-// even for non exposed namespaces, see namespaceExposed.
-func (k *Kubernetes) namespace(n string) bool {
-	ns, err := k.APIConn.GetNamespaceByName(n)
+// filteredNamespaceExists checks if namespace exists in this cluster
+// according to any `namespace_labels` plugin configuration specified.
+// Returns true even for namespaces not exposed by plugin configuration,
+// see namespaceExposed.
+func (k *Kubernetes) filteredNamespaceExists(namespace string) bool {
```

Comment:
Why not simply add this logic to namespaceExposed?
It would avoid a lot of modifications and would probably make the code easier to read?

Reply:
Yeah, i've been toying with something along these lines in my head. Given that `namespaces` and `namespace_labels` are mutually exclusive, I think it should be fine. If they were not mutually exclusive, I believe there are some edge cases that would need to be covered but that is not the case.